### PR TITLE
Support Password Retrieval via SSM

### PR DESCRIPTION
**Pull request by ggotti**
Added support for pulling in password via [AWS Systems Manager Parameter Store](http://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-paramstore.html)
- Utilises SpringCloud to boostrap property retrieval.
- Modified Main Application generator Java class to allow application to start successfully when it cannot resolve the Hostname

### CODE REVIEW

File: generators/more-aws/index.js [hunk 1]
```diff
@@ -27,6 +37,9 @@ module.exports = class extends BaseGenerator {
                 this.baseName = this.config.get(constants.conf.baseName);
                 this.appsFolders = this._getAppFolders();
                 this.directoryPath = './';
+                this.packageName = this.config.get('packageName');
+                this.packageFolder = this.config.get('packageFolder');
+                this.buildTool = this.config.get('buildTool');
```
Comment (carvallegro) [inline generators/more-aws/index.js]: Put these in the constants.js file?

File: generators/more-aws/index.js [hunk 2]
```diff
@@ -181,6 +194,22 @@ module.exports = class extends BaseGenerator {
...
                     });
             },
+            addAWSDependencies() {
+
+                if (this.buildTool === 'maven') {
+                    this.addMavenDependency('com.amazonaws', 'aws-java-sdk-ssm', AWS_SSM_VERSION);
+                    this.addMavenDependency('org.springframework.cloud', 'spring-cloud-context', SPRING_CLOUD_CTX_VERSION);
+                } else if (this.buildTool === 'gradle') {
+                    this.addGradleDependency('compile', 'com.amazonaws', 'aws-java-sdk-ssm', AWS_SSM_VERSION);
+                    this.addGradleDependency('compile', 'org.springframework.cloud', 'spring-cloud-context', SPRING_CLOUD_CTX_VERSION);
```
Comment (carvallegro) [inline generators/more-aws/index.js]: If you've put `AWS_SSM_VERSION` and `SPRING_CLOUD_CTX_VERSION` as constanst, I think you can put the other parameters as well.

File: generators/more-aws/index.js (hunk 1 above)
Comment (ggotti) [reply generators/more-aws/index.js]: This seems to be the way they're doing it in other generators.

File: generators/more-aws/index.js (hunk 2 above)
Comment (ggotti) [reply generators/more-aws/index.js]: Fixed 🌮

**Pull request merged**